rust/src/index/vector/diskann/builder.rs

```diff
@@ -316,7 +313,7 @@ async fn index_once<V: Vertex + Clone>(
             .try_collect::<Vec<_>>()
             .await?;
         for (j, nbs) in neighbours {
-            graph.set_neighbors(j, nbs);
+            graph.set_neighbors(j, Arc::new(nbs.iter().copied().collect::<UInt32Array>()));
```

Comment:
Would `into_iter()` work here? That may avoid the copy (although it might be optimized away implicitly)
```suggestion
            graph.set_neighbors(j, Arc::new(nbs.into_iter().collect::<UInt32Array>()));
```